src/main/java/com/example/webapp/Controller/DepartmentController.java

```diff
@@ -0,0 +1,51 @@
+package com.example.webapp.Controller;
+
+import com.example.webapp.Model.Department;
+import com.example.webapp.Service.DepartmentService;
+import com.example.webapp.dto.DepartmentDTO;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/departments")
+public class DepartmentController {
+
+
+    private final DepartmentService departmentService;
+
+    @GetMapping
+    public ResponseEntity<List<DepartmentDTO>> getDepartments() {
+        List<DepartmentDTO> list = departmentService.getAll();
+        return ResponseEntity.ok(list);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<DepartmentDTO> getDepartmentById(@PathVariable long id) {
+        return ResponseEntity.ok(departmentService.getById(id));
+    }
+
+    @PostMapping
+    public ResponseEntity<DepartmentDTO> addDepartment(@Valid @RequestBody DepartmentDTO department) {
+        departmentService.add(department);
```

Comment:
You can return just Id of created department

Reply:
Done